vllm/engine/metrics.py

```diff
@@ -60,6 +60,22 @@ def __init__(self, labelnames: List[str]):
             name="vllm:generation_tokens_total",
             documentation="Number of generation tokens processed.",
             labelnames=labelnames)
+        self.counter_request_success = Counter(
+            name="vllm:request_success",
+            documentation="Count of successfully processed requests.",
+            labelnames=labelnames)
```

Comment:
A quick option to add http related metrics would be to use [`prometheus-fastapi-instrumentator`](https://github.com/trallnag/prometheus-fastapi-instrumentator/tree/master).

This involves installing the package:
```pip install prometheus-fastapi-instrumentator```

Then adding the following 2 lines after the app creation:
https://github.com/vllm-project/vllm/blob/8af890a865bf9f744d7d9bd5515558b42224c744/vllm/entrypoints/openai/api_server.py#L47-L48
```
from prometheus_fastapi_instrumentator import Instrumentator
Instrumentator().instrument(app).expose(app)
```

This will add the following metrics:

| Metric Name                    | Type     | Description                                                                                                     |
|-------------------------------|----------|-----------------------------------------------------------------------------------------------------------------|
| http_requests_total            | counter  | Total number of requests by method, status, and handler.                                                        |
| http_request_size_bytes        | summary  | Content length of incoming requests by handler. Only value of header is respected. Otherwise ignored.         |
| http_response_size_bytes       | summary  | Content length of outgoing responses by handler. Only value of header is respected. Otherwise ignored.         |
| http_request_duration_highr_seconds | histogram | Latency with many buckets but no API specific labels. Made for more accurate percentile calculations.       |
| http_request_duration_seconds  | histogram | Latency with only a few buckets by handler. Made to be only used if aggregation by handler is important.     |



Should I add it to the PR?

Reply:
I like this solution, it saves us from reinventing the wheel in vLLM

Reply:
done :)

Reply:
Nice, if we're going to be using `prometheus-fasapi-instrumentator` then this implementation should be removed.

Reply:
The code highlighted in the original comment
```python
        self.counter_request_success = Counter(
            name="vllm:request_success",
            documentation="Count of successfully processed requests.",
            labelnames=labelnames)
```
can be removed if we are getting these metrics from `prometheus-fastapi-instrumentator` instead.

Reply:
Thank you for clarifying. I believe that `vllm:request_success` remains valuable. It includes a `finished_reason` label, which allows for counting requests based on their finished reason — either `stop` if the sequence ends with an EOS token, or `length` if the sequence length reaches either `scheduler_config.max_model_len` or `sampling_params.max_tokens`. I'm open to adjusting its name and description to make it more indicative.

Reply:
What do you think of the idea of renaming this to something like `vllm:request_info` and including `n` and `best_of` as `labels` too? This way we log a single metric from which the user can construct many different visualisations on Grafana by utilising the label filters?

Reply:
While the idea of combining these metrics may seem appealing at first glance, I believe they should be kept separate for the following reasons:

1. The metrics have different types: `vllm:request_success` is a Counter, while `vllm:request_params_best_of` and `vllm:request_params_n` are Histograms.

2. Aggregating different labels lacks semantic meaning.

3. Although merging `n` and `best_of` into the same histograms might make sense in this case, as they would share the same buckets, we may encounter scenarios where we need to introduce another metric with different bucket requirements.

4. This situation differs from the `Info` metric type, where data is encoded in the label values.